Release 1.0

boblangley opened the pull request:
Conencts to https://github.com/Particular/DotNetCoreLaunch/issues/27

Pull request to review, and merge, 1.0 for release.

boblangley commented on README.md:
```diff
@@ -1,3 +1,2 @@
-# NServiceBus.SagaAudit
-
+# ServiceControl.Plugin.Nsb6.SagaAudit
 Enables audit of saga state changes for NServiceBus version 6.x endpoints to facilitate the SagaView state change diagram in ServiceInsight.
```
Perhaps this file change should be removed from the branch?

boblangley commented on src/Directory.Build.props:
```diff
@@ -0,0 +1,35 @@
+    DependsOnTargets="GetVersion"
+    Condition="'$(GitVersion_Task_targets_Imported)' == 'True'" />
+
+</Project>
```
Should we skip this and go straight to `Particular.Packaging`?

boblangley commented on src/NServiceBus.SagaAudit.AcceptanceTests/When_a_message_is_handled_by_a_saga.cs:
```diff
@@ -0,0 +1,171 @@
+﻿namespace ServiceControl.Plugin.Nsb6.SagaAudit.AcceptanceTests
```
This is nit-picky, but should we kill the `Nsb6` portion of the namespace wherever we are using it?

WilliamBZA commented on src/NServiceBus.SagaAudit.AcceptanceTests/EndpointTemplates/ContextAppender.cs:
```diff
@@ -0,0 +1,129 @@
+            return this;
+        }
+
+        public bool IsDebugEnabled { get{return true;}}
```
Nitpick: Some weird spacing around here

WilliamBZA commented on src/NServiceBus.SagaAudit/CaptureSagaResultingMessagesBehavior.cs:
```diff
@@ -0,0 +1,66 @@
+                    ResultingMessageId = context.OutgoingMessage.Id,
+                    TimeSent = DateTimeExtensions.ToUtcDateTime(context.OutgoingMessage.Headers[Headers.TimeSent]),
+                    MessageType = logicalMessage.MessageType.ToString(),
+                    //TODO: Can we remove the DeliveryDelay and DeliveryAt here??
```
Can we?

WilliamBZA commented on src/NServiceBus.SagaAudit/CaptureSagaStateBehavior.cs:
```diff
@@ -0,0 +1,147 @@
+
+        public SagaChangeInitiator BuildSagaChangeInitatorMessage(Dictionary<string, string> headers, string messageId, string messageType )
+        {
+
```
Seems weird to have a leading blank line here

WilliamBZA commented on src/NServiceBus.SagaAudit/SagaAuditFeature.cs:
```diff
@@ -0,0 +1,53 @@
+        
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            context.Settings.TryGet("NServiceBus.SagaAudit.Serialization", out Func<object, Dictionary<string, string>> customSagaEntitySerialization);
```
Side note: I LOVE this new syntax.

Actual comment: Could the `TryGet` ever return false? What happens if `customSagaEntitySerialization` is null?

WilliamBZA commented on src/NServiceBus.SagaAudit/ServiceControlBackend.cs:
```diff
@@ -0,0 +1,78 @@
+        public Address Destination { get; set; }
+        public Address LocalAddress { get; set; }
+
+        public  ServiceControlBackend(ISendMessages messageSender)
```
Nitpick: Double space here

SzymonPobiega commented:
@boblangley I removed all references to the old namespace

SzymonPobiega commented:
@boblangley I fixed the namespaces and package description. Should be clean now.

boblangley commented on src/NServiceBus.SagaAudit/SagaEntitySerializationStrategy.cs:
```diff
@@ -0,0 +1,75 @@
+        {
+            return t.IsPrimitive || t == typeof(string) || t == typeof(Guid) || t == typeof(TimeSpan) || t == typeof(DateTime) ||
+                   (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Nullable<>));
+        }
```
@SzymonPobiega I just noticed the test you added to make sure nested types are _not_ serialized. Is this a change in behavior from the current plugins? Looking at them it looks like we just let the JSON serializer do it's thing, without restrictions.

boblangley commented:
@SzymonPobiega My apologies for not posting this:

When we use Particular.Packaging and also explicitly bring in GitVersionTask as a dependency we do not need to do any of the workarounds. I rebased the branch and removed those commits and adjusted 2.0/3.0 accordingly.

I smoke tested the beta with ServiceInsight. All of our tests and acceptance tests are passing. I am merging this PR and staging 1.0 in octopus.
